src/streaming-client.ts

```diff
@@ -94,12 +98,19 @@ export class RevAiStreamingClient extends EventEmitter {
         return new BufferedDuplex(this.requests, this.responses, { readableObjectMode: true });
     }
 
+    /**
+     * Signals to the api that you have finished sending data.
+     */
+    public finish(): void {
+        this.requests.end('EOS', 'utf8');
+    }
+
     /**
      * Ends the streaming connection and closes off the buffer returned from start()
      */
     public end(): void {
```

Comment:
what's the different between the `finish()` and `end()` methods?

In `finish` you do `this.requests.end('EOS', 'utf8') but in `end`, you do just `this.request.end()` with no `EOS` or `utf8`

Reply:
Finish signals to the server that the client is done giving input. End immediately cuts off the connection and stops everything

Reply:
Maybe we can combine the two so that there is only end which sends EOS and ends the connection . Why should we keep them separate?

Reply:
The difference is that a user may not want to end the connection immediately.

If anything I think we should remove the end method since it technically breaks the contract we have (the user shouldn't be the one to end the connection)